Add custom css width utils to work arrays of components (add skeletons correction)

fishtriangle opened the pull request:
add file utils.css to add width utils
globals.css - add import of utils and add variable of width step CategoryView.tsx - change w utility with width utility

All other changes - according dprint formatter rules

fishtriangle committed:
Add custom css width utils to work arrays of components

add file utils.css to add width utils
globals.css - add import of utils and add variable of width step
CategoryView.tsx - change w utility with width utility

All other changes - according dprint formatter rules

fishtriangle committed:
Merge branch 'main' into Add-feature-skeletons

Loskir commented on styles/globals.css:
```diff
@@ -1,9 +1,14 @@
 @tailwind utilities;
 
+:root {
+    --width: 8.333333333333333%;
```
Very strange global variable

Loskir commented on styles/globals.css:
```diff
@@ -1,9 +1,14 @@
 @import url('https://fonts.googleapis.com/css2?family=Inter:wght@400;500;600;700&display=swap');
+@import "utils.css";
```
I'd prefer it not to be global

Loskir reviewed (changes requested):
Maybe calculate widths in pixels and then apply them with react styles?

fishtriangle committed:
Add custom css width prop to BaseSkeleton

BaseSkeleton.tsx - add optional prop width
CategoryView.tsx - change width assignment
globals.css - delete unused variable and import
delete utils.css as unused

fishtriangle committed:
Merge remote-tracking branch 'origin/Add-feature-skeletons' into Add-feature-skeletons

Loskir commented on components/Skeletons/BaseSkeleton.tsx:
```diff
@@ -1,16 +1,22 @@
           className,
           "bg-gray-100 rounded-lg relative before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white before:to-transparent isolate overflow-hidden",
         )}
+        style={width ? { width: `${width}px` } : {}}
```
Maybe pass the style object itself?

fishtriangle committed:
BaseSkeleton component correction

BaseSkeleton.tsx - change style appointment

Loskir commented on components/Skeletons/BaseSkeleton.tsx:
```diff
@@ -3,7 +3,7 @@ import React from "react"
 interface BaseSkeletonProps {
   className?: string
-  width?: number
+  width?: string | number
```
No, I mean
`style?: CSSProperties`
